src/Workspaces/Core/Portable/Workspace/Host/TextFactory/ITextFactoryService.cs

```diff
@@ -28,5 +28,15 @@ internal interface ITextFactoryService : IWorkspaceService
         /// </exception>
         /// <exception cref="IOException">An IO error occurred while reading from the stream.</exception>
         SourceText CreateText(Stream stream, Encoding defaultEncoding, CancellationToken cancellationToken = default(CancellationToken));
+
+
+        /// <summary>
+        /// Creates <see cref="SourceText"/> from a reader.
+        /// </summary>
+        /// <param name="reader">The <see cref="TextReader"/> to read the text from.</param>
+        /// <param name="encoding">Specifies an encoding for the <see cref="SourceText"/>SourceText. it could be even null</param>
```

Comment:
no, there is no encoding associated with TextReader. Encoding is something that is used to convert text to stream and via versa. TextReader abstract that out, so consumer of the TextReader doesn't need to care about Encoding.

the encoding given is for SourceText which will be used later when we convert sourcetext to stream. (ex, to calculate checksum or write to a file and etc)